examples/function-pointers/function_pointers.bal

```diff
@@ -1,41 +1,44 @@
 import ballerina/io;
 import ballerina/lang.'int;
 
-// The `test` function acts as a variable function pointer in the `main` function.
-function test(string s, int... x) returns float {
-    int|error y = 'int:fromString(s);
-    float f = 0.0;
+public function main() {
+    // The function name `test` serves as a function pointer argument in the
+    // call to `foo()`. Function names can be thought of as final variables
+    // since although you can use it like a regular variable, you cannot
+    // modify the value it's associated with.
```

Comment:
```suggestion
    // modify the value it is associated with.
```

Reply:
Do we need to say this specifically though?

Reply:
Added for the sake of completeness, since we're drawing parallels with variables.